sys/reboot/src/log_reboot.c

```diff
@@ -383,6 +384,50 @@ reboot_conf_export(void (*func)(char *name, char *val),
     return 0;
 }
 
+void
+log_reboot_reason_str(enum hal_reset_reason reason, char *buf)
+{
+    char str_reason[MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE)];
+
+    if (reason >= HAL_RESET_OTHER) {
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"OTHER: 0x%X",reason - HAL_RESET_OTHER);
+        strcpy(buf, str_reason);
+        return;
+    }
+
+    switch (reason) {
+    case HAL_RESET_POR:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"HARD");
+        break;
+    case HAL_RESET_PIN:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"RESET_PIN");
+        break;
+    case HAL_RESET_WATCHDOG:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"WDOG");
+        break;
+    case HAL_RESET_SOFT:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"SOFT");
+        break;
+    case HAL_RESET_BROWNOUT:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"BROWNOUT");
+        break;
+    case HAL_RESET_REQUESTED:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"REQUESTED");
+        break;
+    case HAL_RESET_SYS_OFF_INT:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"SYSTEM_OFF_INT");
+        break;
+    case HAL_RESET_DFU:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"DFU");
+        break;
+    default:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"UNKNOWN");
```

Comment:
would be better to stringify this like `"UNKNOWN (%d)"` so actual value is also logged